kodi_addon_checker/schema_validation.py

```diff
@@ -0,0 +1,33 @@
+import xmlschema
+import logging
+
+from .report import Report
+from .record import PROBLEM, Record, INFORMATION
+
+xml_schema = "./xml_schema/"
+LOGGER = logging.getLogger(__name__)
+
+
+def schemas(report: Report, parsed_xml):
+
+    for extension in parsed_xml.findall("extension"):
+        extension_point = (extension.get("point")).split(".")[-1]
+        schema_path = xml_schema + extension_point + '.xsd'
+        validate(report, extension, schema_path)
+        parsed_xml.remove(extension)
+
+    root_schema = xml_schema + 'addon.xsd'
+    validate(report, parsed_xml, root_schema)
+
+
+def validate(report: Report, xml, schemapath):
+
+    schema = xmlschema.XMLSchema(schemapath)
+
+    try:
+        schema.validate(xml)
+        report.add(Record(INFORMATION, "Valid XML file found"))
+
+    except xmlschema.XMLSchemaValidationError as schemaerror:
+        LOGGER.error(schemaerror)
+        report.add(Record(PROBLEM, "Schema validation failed"))
```

Comment:
report the reason, otherwise you have to guess what's wrong

Reply:
Log file doesn't help as you won't see that on travis.

Reply:
I did that to keep the travis log clean. If you want I'll add the report the reason to normal logs.